packages/react-magma-dom/src/components/ToggleButton/ToggleButton.tsx

```diff
@@ -0,0 +1,231 @@
+import { transparentize } from 'polished';
+import { XOR } from '../../utils';
+
+import { ThemeContext } from '../../theme/ThemeContext';
+import { ThemeInterface } from '../../theme/magma';
+import { Button, ButtonProps, ButtonSize } from '../Button';
+import { ButtonColor } from '../Button';
+import { IconButton } from '../IconButton';
+import { IconProps } from 'react-magma-icons';
+import { ToggleButtonGroupContext } from './ToggleButtonGroup';
+
+export interface ToggleButtonTextProps
+  extends Omit<ButtonProps, 'isChecked' | 'isDefault'> {
+  /**
+   * Enables a specified button to be active.
+   */
+  defaultChecked?: boolean;
+  /**
+   * Sets a disabled state for a button.
+   */
+  disabled?: boolean;
+  /**
+   * Icon which displays alongside text.
+   */
+  icon?: React.ReactElement<IconProps>;
+  isChecked?: boolean;
+  isDefault?: boolean;
+  isInverse?: boolean;
+  /**
+   * Restricts deselection of a button within a single selection group.
+   */
+  requiredSelect?: boolean;
+  /**
+   * Sets a single selection group.
+   */
+  singleSelect?: boolean;
+  /**
+   * Changes the button size throughout the group between 'small', 'medium', and 'large'.
+   */
+  size?: ButtonSize;
+  /**
+   * @internal
+   */
+  theme?: ThemeInterface;
+}
+
+export interface ToggleButtonIconProps
+  extends Omit<
+    ButtonProps,
+    'children' | 'isChecked' | 'isDefault' | 'hasLabel'
+  > {
+  /**
+   * Sets the aria-label which is required for icon only buttons.
+   */
+  'aria-label': string;
+  /**
+   * Enables a specified button to be active.
+   */
+  defaultChecked?: boolean;
+  /**
+   * Sets a disabled state for a button.
+   */
+  disabled?: boolean;
+  hasLabel?: boolean;
+  /**
+   * Sets a single selection group.
+   */
+  singleSelect?: boolean;
+  /**
+   * Icon which displays alongside text.
+   */
+  icon: React.ReactElement<IconProps>;
+  isChecked?: boolean;
+  isDefault?: boolean;
+  isInverse?: boolean;
+  /**
+   * Restricts deselection of a button within a single selection group.
+   */
+  requiredSelect: boolean;
+  /**
+   * Changes the button size throughout the group between 'small', 'medium', and 'large'.
+   */
+  size?: ButtonSize;
+}
+
+export type ToggleButtonProps = XOR<
+  ToggleButtonTextProps,
+  ToggleButtonIconProps
+>;
+
+//Sets the icon width for icon only Toggle Buttons.
+export function setIconWidth(props: ToggleButtonProps) {
+  if (props.size === ButtonSize.small) {
+    return props.theme.spaceScale.spacing07;
+  }
+  if (props.size === ButtonSize.medium) {
+    return props.theme.spaceScale.spacing09;
+  }
+  if (props.size === ButtonSize.large) {
+    return props.theme.spaceScale.spacing11;
+  }
+  return props.theme.spaceScale.spacing09;
+}
+
+//Sets the background colors for the Toggle Buttons.
+export function setBackgroundColor(props: ToggleButtonProps) {
+  if (
+    (props.isChecked && props.defaultChecked) ||
+    (!props.isChecked && props.isDefault)
+  ) {
+    return '';
```

Comment:
Maybe we return `'none'` as we are returning a background color instead of blank?